ui/app/components/send/to-autocomplete.js

```diff
@@ -2,54 +2,126 @@ const Component = require('react').Component
 const h = require('react-hyperscript')
 const inherits = require('util').inherits
 const Identicon = require('../identicon')
+const AccountListItem = require('./account-list-item')
 
 module.exports = ToAutoComplete
 
 inherits(ToAutoComplete, Component)
 function ToAutoComplete () {
   Component.call(this)
+
+  this.state = { accountsToRender: [] }
+}
+
+ToAutoComplete.prototype.getListItemIcon = function (listItemAddress, toAddress) {
+  const listItemIcon = h(`i.fa.fa-check.fa-lg`, { style: { color: '#02c9b1' } })
+
+  return toAddress && listItemAddress === toAddress
+    ? listItemIcon
+    : null
+}
+
+ToAutoComplete.prototype.renderDropdown = function () {
+  const {
+    accounts,
+    closeDropdown,
+    onChange,
+    to,
+  } = this.props
+  const { accountsToRender } = this.state
+
+  return accountsToRender.length && h('div', {}, [
+
+    h('div.send-v2__from-dropdown__close-area', {
+      onClick: closeDropdown,
+    }),
+
+    h('div.send-v2__from-dropdown__list', {}, [
+
+      ...accountsToRender.map(account => h(AccountListItem, {
+        account, 
+        handleClick: () => {
+          onChange(account.address)
+          closeDropdown()
+        }, 
+        icon: this.getListItemIcon(account.address, to),
+        displayBalance: false,
+        displayAddress: true,
+      }))
+
+    ]),
+
+  ])
+}
+
+ToAutoComplete.prototype.handleInputEvent = function (event = {}, cb) {
+  const {
+    to,
+    accounts,
+    closeDropdown,
+    openDropdown,
+  } = this.props
+
+  const matchingAccounts = accounts.filter(({ address }) => address.match(to))
+
+  if (!to) {
```

Comment:
This logic seems complicated. I am suspecting that there is a more straight-forward  to handle it. Can you explain what this is trying to do?

Reply:
I've simplified the logic in 99f32048

The logic handles the following cases:

- If there is no text entered in the To input field (`to == false`), then show all accounts in the dropdown
- If the text entered in the To input field exactly matches a single address , then close the dropdown and highlight the text in the input field (to indicate to the user that it can be edited)
- If the text entered in the To input field matches a substring of one or more of the addresses of the accounts, show those accounts with matching addresses in an open dropdown,
- If none of the above are true, it means that there is some text entered in the To input field, but that the text does not match any of the account addresses, close the dropdown and highlight the input text